backend/src/graphql/resolver/TransactionResolver.ts

```diff
@@ -289,64 +228,90 @@ export class TransactionResolver {
   ): Promise<TransactionList> {
     // load user
     const userRepository = getCustomRepository(UserRepository)
-    let userEntity: dbUser | undefined
-    if (userId) {
-      userEntity = await userRepository.findOneOrFail({ id: userId }, { withDeleted: true })
-    } else {
-      userEntity = await userRepository.findByPubkeyHex(context.pubKey)
+    const user = userId
+      ? await userRepository.findOneOrFail({ id: userId }, { withDeleted: true })
+      : await userRepository.findByPubkeyHex(context.pubKey)
+    let limit = pageSize
+    let offset = 0
+    let skipFirstTransaction = false
+    if (currentPage > 1) {
+      offset = (currentPage - 1) * pageSize - 1
+      limit++
     }
-
-    const transactions = await listTransactions(
-      currentPage,
-      pageSize,
+    if (offset && order === Order.ASC) {
+      offset--
+    }
+    const userTransactionRepository = getCustomRepository(UserTransactionRepository)
+    let [userTransactions, userTransactionsCount] = await userTransactionRepository.findByUserPaged(
+      user.id,
+      limit,
+      offset,
       order,
-      userEntity,
       onlyCreations,
     )
+    skipFirstTransaction = userTransactionsCount > offset + limit
```

Comment:
This looks scary.

Reply:
![image](https://user-images.githubusercontent.com/1238238/154942851-9fedcb2d-0985-4005-bce8-3790d4f2ce02.png)